src/status_im/ui/screens/wallet/sign_message/views.cljs

```diff
@@ -103,7 +103,7 @@
           nil]]]]
       [enter-password-buttons value-atom false
        #(re-frame/dispatch [:wallet/discard-transaction-navigate-back])
-       #(re-frame/dispatch [:wallet/sign-message typed? (merge screen-params @value-atom)])
+       #(re-frame/dispatch [:wallet/sign-message typed? (merge screen-params @value-atom) value-atom])
```

Comment:
Interesting, I am all for local state when actually local and well encapsulated, but in this case looks like it's not local anymore, and relies on a state modification happening in a separate component/file (state is being passed around, but modified directly i.e `swap`), at this point I would just consider using the re-frame database, as there's no encapsulation effectively anymore.

Maybe is just a matter of a bit of refactoring (`views.cljs` never access state directly, but always access it through `events.cljs`, i.e `(events/wrong-password?)` `(events/sign-message! typed? screen-params)` so that `value-atom` never leaves `events.cljs` directly, but always through an API.

Not much of a big deal to be honest, but it's good to agree on a direction, what do you think?

Reply:
I agree with your point, although I'd rather fix it in a different way - pass a callback from the view code e.g. `(show-wrong-password)`. That callback can alter the state and then it's local i.e. encapsulated.

Whether or not there's a password error really is UI state and nothing more. It doesn't belong in the events code and it doesn't belong in re-frame database.

Reply:
callback sounds like a good approach, best to keep the atom in the view I agree :+1: